### PR TITLE
ostree: extract `httpClientForRef` from resolveRef

### PR DESCRIPTION
**Pull request by mvo5**
This commit extracts a `httpClientForRef` helper from the `resolveRef()`
function. This helps with readability and testability. It also adds
a test for the proxy setting.

Not ideal as the test knows a bit too much about the implementation details but OTOH realtively straightfoward to add.

[build on top of https://github.com/osbuild/images/pull/1050]

### CODE REVIEW

**Comment by lzap** [comment]
I think this is great, can you add a test for client cert/key as well as CA certs?

**Comment by mvo5** [comment]
> I think this is great, can you add a test for client cert/key as well as CA certs?

Added

File: pkg/ostree/ostree_test.go [hunk 1]
```diff
@@ -226,3 +227,54 @@ func TestValidate(t *testing.T) {
...
+	}
+	client, err := httpClientForRef("https", ss)
+	assert.NoError(t, err)
+
+	tlsConf := client.Transport.(*http.Transport).TLSClientConfig
+	assert.NoError(t, err)
+	// checking this from the outside is hard, there is no way to
+	// poke at the RootCAs internals so we just ensure one got added
+	assert.Equal(t, 1, len(tlsConf.RootCAs.Subjects()))
+	// no certificates got added
```
Comment (mvo5) [inline pkg/ostree/ostree_test.go:278]: Hm, this lint error is unfortunate, AFAICT this is the only way to introspect a golang CertPool, I hope that before they remove it they will also add something to introspect again, it seems strange

**Review by bcl — APPROVED** [review]
Looks good to me!

**Review by supakeen — APPROVED** [review]

**Review by lzap — APPROVED** [review]
Thank you.

**Pull request merged**